lib/Ohmbrewer_Relay.cpp

```diff
@@ -1,36 +1,89 @@
 #include "Ohmbrewer_Relay.h"
 #include "Ohmbrewer_Screen.h"
+#include "Ohmbrewer_Publisher.h"
+
 
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param powerPin - The power pin - on/off line. Digital pin number X.
- * @param controlPin - The Control pin - Data/speed/power level Digital pin number X.
  */
-Ohmbrewer::Relay::Relay(int id, int powerPin, int controlPin) : Ohmbrewer::Equipment(id) {
+Ohmbrewer::Relay::Relay(int id, int pumpPin) : Ohmbrewer::Equipment(id) {
+
+    _powerPin = pumpPin;
+    _controlPin = -1;
+}
+
+/**
+ * Constructor
+ * @param id The Sprout ID to use for this piece of Equipment
+ * @param relayPins - controlPin always first in <list>
+ *  controlPin - The Control pin - Data/speed/power level Digital pin number X.
+ *  powerPin - The power pin - on/off line. Digital pin number X.
+ */
+Ohmbrewer::Relay::Relay(int id, std::list<int>* relayPins) : Ohmbrewer::Equipment(id) {
+    int size = relayPins->size();
+    int powerPin = -1;
+    int controlPin = -1;
+    if  (size == 2){ // switch and relay
+        controlPin = relayPins->front();
+        powerPin = relayPins->back();
+    }else if (size == 1){ //relay only
+        controlPin = relayPins->front();
+    }else {//publish error
+        Ohmbrewer::Publisher::publish_map_t pMap;
+        Ohmbrewer::Publisher* pub = new Ohmbrewer::Publisher(new String("error_log"), &pMap);
+        pMap[String("list_check_relay")] = String("improperly formed input - Relay(int, int<list>)");
+        pub->publish();
+    }
+    //will be set to -1 for error checking if not enabled
     _powerPin = powerPin;
     _controlPin = controlPin;
-    // For now, we will not automatically add a Spark.function to Relays as
-    // it's used mostly as a base class and our subclasses call the Relay constructor. If we can find a safe way to
-    // determine if a function for actual Relay types should be added, then we'll change that.
 }
 
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param powerPin - The power pin - on/off line. Digital pin number X.
- * @param controlPin - The Control pin - Data/speed/power level Digital pin number X.
+ * @param pumpPin - Single speed pump will only have PowerPin
```

Comment:
Along with the comment below, fix this to something like "the updated power pin" or some such.